### PR TITLE
Int64 abstract

### PR DESCRIPTION
**Pull request by Simn**
Tests aren't done yet but I wanted to open the PR before heading to bed!

@hughsando: I have disabled your Int64.hx for now so we can look into integrating it with the new abstract. We will have to make sure that the API is compatible and that documentation generation still works.

### CODE REVIEW

File: std/haxe/Int64.hx [hunk 1]
```diff
@@ -99,176 +134,308 @@ class Int64 {
...
+		Returns the modulus of `a` divided by `b`.
+	**/
+	@:op(A % B) public static inline function mod( a : Int64, b : Int64 ) : Int64
+		return divMod(a, b).modulus;
 
-	public static inline function shr( a : Int64, b : Int ) : Int64 {
-		return if( b & 63 == 0 ) a else if( b & 63 < 32 ) new Int64( a.high >> b, ushr32(a.low,b) | (a.high << i32(32 - (b&63))) ) else new Int64( a.high >> 31, a.high >> i32(b - 32) );
-	}
+	@:op(A % B) private static inline function modInt( a : Int64, b : Int ) : Int64
+		return mod( a, b ).toInt();
```
Comment (waneck) [inline std/haxe/Int64.hx]: Was there a reason for those `toInt()`?

**Comment by ncannasse** [comment]
Any reason to hold on merging this ?
Also, a last-moment API change require: could we change  getLow()/getHigh() by simply reading .low/.high (no setter) ?

**Comment by Simn** [comment]
This is merged already.

**Comment by ncannasse** [comment]
Oops, right. Can we make the change for low/high ? (and @:deprecated getLow/getHigh) ?

**Comment by Simn** [comment]
I'm not sure if this is worth deprecating over but I don't have any objections either.

**Comment by Simn** [comment]
There's this though: https://github.com/HaxeFoundation/haxe/blob/development/std/haxe/io/FPHelper.hx#L220

**Comment by ncannasse** [comment]
@Simn we could unsafe cast to ___Int64 in that case

**Comment by Simn** [comment]
I'll just keep `set_low` around as a private function, but still use `var low(get, never)`. This way we can call the setter manually without having to cast (which might introduce weird field access).

**Comment by ncannasse** [comment]
@Simn fine of me, as long as var low is public ;)

**Comment by Simn** [comment]
Yeah, done.

**Comment by ckentgeorge** [comment]
Is there a reason why we don't allow setting high and low directly?  i.e. var low(get, set)?

**Comment by ncannasse** [comment]
@ckentgeorge on some platforms Int64 might be immutable, the setter are only used for some platforms when we want to directly mutate because we know it's allowed in that specific case

**Pull request merged**